### PR TITLE
Add class xml comment with supported rule IDs

### PR DESCRIPTION
**Pull request by Evangelink**
This simplifies finding the class implementing a specific rule by allowing Ctrl + , search.

cc @sharwell as you started this work some time ago.

### CODE REVIEW

**Comment by Evangelink** [comment]
@mavasani Do you have some time to give your opinion here?

File: src/Microsoft.CodeAnalysis.Analyzers/CSharp/CSharpImmutableObjectMethodAnalyzer.cs [hunk 1]
```diff
@@ -13,6 +13,9 @@ namespace Microsoft.CodeAnalysis.CSharp.Analyzers
 {
     using static CodeAnalysisDiagnosticsResources;
 
+    /// <summary>
+    /// RS1014: <inheritdoc cref="DoNotIgnoreReturnValueOnImmutableObjectMethodInvocationTitle"/>
```
Comment (mavasani) [inline src/Microsoft.CodeAnalysis.Analyzers/CSharp/CSharpImmutableObjectMethodAnalyzer.cs:17]: How does this help in Ctrl + , search? It does improve the quickinfo for the type names, which is valuable by itself, but wanted to know if there is some prefix I should add in Ctrl + , search for either the rule ID or the content/name of the title string to lead to results.

**Review by mavasani — APPROVED** [review]
LGTM.

I did not verify every file that the rule ID/title name you added in the doc comment matches the supported descriptors for corresponding analyzer, I trust your own verification - let me know if you'd prefer me to closely review all changes.

File: src/Microsoft.CodeAnalysis.Analyzers/CSharp/CSharpImmutableObjectMethodAnalyzer.cs (hunk 1 above)
Comment (Evangelink) [reply src/Microsoft.CodeAnalysis.Analyzers/CSharp/CSharpImmutableObjectMethodAnalyzer.cs:17]: You can do `Ctrl+,` and then type `CAXXXX` (`RS1014` in this case) and the file will appear in the list. I don't know the details of how the search is working. @sharwell Started this change on the codebase some time ago but it seems that it wasn't fully done or it got lost in refactorings/new rules addition. The rule explanation (the `<inheritdoc>` part) is not so helpful so we could remove it. With current way, if you call the type you see in the intellisense doc rule ID + rule title.

**Comment by Evangelink** [comment]
I did full review so it should be ok.

**Comment by mavasani** [comment]
Yes, please.

**Pull request merged**